### PR TITLE
[refactor] refactoring code to remove duplicate code

### PR DESCRIPTION
**Pull request by yandongxiao**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 
 # Image URL to use all building/pushing image targets
 # ENVTEST_K8S_VERSION refers to the version of kubebuilder assets to be downloaded by envtest binary.
-ENVTEST_K8S_VERSION = 1.23
+ENVTEST_K8S_VERSION = 1.27
```
Comment (kevincai) [inline Makefile:4]: right now, we are promising the compatibility of 1.19 ~ 1.25, will this version too high for the compatibility?

File: config/crd/bases/starrocks.com_starrocksclusters.yaml [hunk 2]
```diff
@@ -1064,7 +1064,7 @@ spec:
                       type: object
                     type: array
                   image:
-                    description: Image for a starrocks be deployment.
+                    description: Image for a starrocks deployment..
```
Comment (kevincai) [inline config/crd/bases/starrocks.com_starrocksclusters.yaml:1067]: redundant dot

**Review by kevincai — COMMENTED** [review]
TBD

File: Makefile (hunk 1 above)
Comment (yandongxiao) [reply Makefile:4]: this modification is by mistake

File: pkg/k8sutils/templates/pod/spec.go [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+				FieldRef: &corev1.ObjectFieldSelector{FieldPath: "metadata.namespace"},
+			},
+		},
+		{
+			Name:  "HOST_TYPE",
+			Value: "FQDN",
+		},
+		{
+			Name:  "USER",
+			Value: "root",
```
Comment (kevincai) [inline pkg/k8sutils/templates/pod/spec.go:196]: // TODO: make this configurable.

File: config/crd/bases/starrocks.com_starrocksclusters.yaml (hunk 2 above)
Comment (yandongxiao) [reply config/crd/bases/starrocks.com_starrocksclusters.yaml:1067]: done

File: pkg/apis/starrocks/v1/component_type.go [hunk 4]
```diff
@@ -0,0 +1,390 @@
...
+	}
+	return spec.StarRocksComponentSpec.GetServiceName()
+}
+
+func (spec *StarRocksFeSpec) GetStorageVolumes() []StorageVolume {
+	return spec.StorageVolumes
+}
+
+func (spec *StarRocksBeSpec) GetStorageVolumes() []StorageVolume {
+	return spec.StorageVolumes
```
[the remark this answers is not in the document]
Comment (yandongxiao) [reply pkg/apis/starrocks/v1/component_type.go:357]: This is a very good question! Currently, although there is no check for whether `spec *StarRocksBeSpec` is nil, the SubController, during reconciling, will first verify if their respective Spec is nil.

File: pkg/k8sutils/templates/pod/spec.go (hunk 3 above)
Comment (yandongxiao) [reply pkg/k8sutils/templates/pod/spec.go:196]: So, USER==root is its default value?

Comment (kevincai) [reply pkg/k8sutils/templates/pod/spec.go:196]: yes, I mean add a TODO here, we will soon go back here to support non-root mode.

File: pkg/apis/starrocks/v1/component_type.go [hunk 5]
```diff
@@ -0,0 +1,390 @@
...
+	//+optional
+	StorageVolumes []StorageVolume `json:"storageVolumes,omitempty"`
+
+	//+optional
+	//beEnvVars is a slice of environment variables that are added to the pods, the default is empty.
+	BeEnvVars []corev1.EnvVar `json:"beEnvVars,omitempty"`
+}
+
+// StarRocksCnSpec defines the desired state of cn.
+type StarRocksCnSpec struct {
```
[the remark this answers is not in the document]
Comment (yandongxiao) [reply pkg/apis/starrocks/v1/component_type.go:63]: done

File: pkg/apis/starrocks/v1/component_type.go [hunk 6]
```diff
@@ -0,0 +1,390 @@
...
+
+	//serviceAccount for access cloud service.
+	ServiceAccount string `json:"serviceAccount,omitempty"`
+
+	//A special supplemental group that applies to all containers in a pod.
+	// Some volume types allow the Kubelet to change the ownership of that volume
+	// to be owned by the pod:
+	FsGroup *int64 `json:"fsGroup,omitempty"`
+
+	//Replicas is the number of desired Pod, the number is 1,3,5
```
[the remark this answers is not in the document]
Comment (yandongxiao) [reply pkg/apis/starrocks/v1/component_type.go]: done

**Comment by yandongxiao** [comment]
I have submitted a new commit based on the feedback provided in the review above.

**Review by kevincai — APPROVED** [review]

**Pull request merged**